src/devices/imagedev/midiin.cpp

```diff
@@ -18,6 +18,28 @@
 
 DEFINE_DEVICE_TYPE(MIDIIN, midiin_device, "midiin", "MIDI In image device")
 
+INPUT_PORTS_START(midiin)
+	PORT_START("CFG")
+	PORT_CONFNAME(0xff, 0xff, "MIDI In channel")
+	PORT_CONFSETTING(   0xff, "All")
+	PORT_CONFSETTING(   0x00, "1")
+	PORT_CONFSETTING(   0x01, "2")
+	PORT_CONFSETTING(   0x02, "3")
+	PORT_CONFSETTING(   0x03, "4")
+	PORT_CONFSETTING(   0x04, "5")
+	PORT_CONFSETTING(   0x05, "6")
+	PORT_CONFSETTING(   0x06, "7")
+	PORT_CONFSETTING(   0x07, "8")
+	PORT_CONFSETTING(   0x08, "9")
+	PORT_CONFSETTING(   0x09, "10")
+	PORT_CONFSETTING(   0x0a, "11")
+	PORT_CONFSETTING(   0x0b, "12")
+	PORT_CONFSETTING(   0x0c, "13")
+	PORT_CONFSETTING(   0x0d, "14")
+	PORT_CONFSETTING(   0x0e, "15")
+	PORT_CONFSETTING(   0x0f, "16")
+INPUT_PORTS_END
+
```

Comment:
In purely practical terms, this should be put in an anonymous namespace.

In terms of user experience, the description here is misleading.  If I saw this, I would assume this would filter MIDI input, only allowing the selected channel to pass through.  What it actually does is rewrite the channel number.  It needs a better name.

Reply:
Standard MIDI terminology for this would be to call it "MIDI (file) mode" and have "Multi" instead of "All" and "Poly: Channel 1" ... "Poly: Channel 16".

Reply:
Yeah, I wasn't sure what the best wording would be either. I just went with RB's suggestion instead.